src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/CreateGithubIssueTest.cs

```diff
@@ -0,0 +1,149 @@
+using Moq;
+using Octokit;
+using Xunit;
+
+namespace DependencyUpdateErrorProcessor.Tests
+{
+    public class CreateGithubIssueTest : DependencyUpdateErrorProcessorTests
+    {
+        [Theory]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "ProcessPendingUpdatesAsync", "no arguments" , "2200/1/1" , false)]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "UpdateAssetsAsync", "[\"ee8cdcfb-ee51-4bf3-55d3-08d79538f94d\",\"UpdateAssetsAsync\"]",  "2200/1/1", false)]
+        public async Task ShouldCreateIssue(string repoUrl, string branch , string method, string arguments , DateTime errorOccurredAt, bool success)
+        {
+            Mock<RepositoryBranchUpdateHistoryEntry> repositoryBranchUpdate =
+                new Mock<RepositoryBranchUpdateHistoryEntry>();
+            repositoryBranchUpdate.Object.Repository = repoUrl;
+            repositoryBranchUpdate.Object.Branch = branch;
+            repositoryBranchUpdate.Object.Method = method;
+            repositoryBranchUpdate.Object.Timestamp = errorOccurredAt;
+            repositoryBranchUpdate.Object.Arguments = arguments;
+            repositoryBranchUpdate.Object.Success = success;
+            Mock<Octokit.Repository> repository = new Mock<Repository>();
+            GithubClient.Setup(x => x.Repository.Get(It.IsAny<string>(), It.IsAny<string>())).ReturnsAsync(repository.Object);
+            Mock<Maestro.Data.Models.Subscription> subscription = new Mock<Maestro.Data.Models.Subscription>();
+            subscription.Object.Id = Guid.Parse("ee8cdcfb-ee51-4bf3-55d3-08d79538f94d");
+            Context.Subscriptions.Add(subscription.Object);
+            Context.SaveChanges();
+            Mock<Octokit.Issue> issue = new Mock<Issue>();
+            GithubClient.Setup(x => x.Issue.Create(It.IsAny<long>(),It.IsAny<NewIssue>())).ReturnsAsync(issue.Object);
+            Context.RepoBranchUpdateInMemory = new List<RepositoryBranchUpdateHistoryEntry>
+                {repositoryBranchUpdate.Object};
+            DependencyUpdateErrorProcessor errorProcessor =
+                ActivatorUtilities.CreateInstance<DependencyUpdateErrorProcessor>(Scope.ServiceProvider,
+                    Context);
+            await errorProcessor.ProcessDependencyUpdateErrorsAsync();
+            GithubClient.Verify(x => x.Issue.Create(It.IsAny<long>(), It.IsAny<NewIssue>()), Times.Once);
+        }
+
+
+        [Theory]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "SynchronizePullRequestAsync", "no arguments", "2200/1/1", false)]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "UpdateAssetsAsync", "[\"ee8cdcfb-ee51-4bf3-55d3-08d79538f94d\",\"UpdateAssetsAsync\"]", "2200/1/1", false)]
+        public async Task ShouldNotCreateIssue(string repoUrl, string branch, string method, string arguments, DateTime errorOccurredAt, bool success)
```

Comment:
The title of the test should probably indicate _why_ it shouldn't create an issue.  Because if this fails, and it does create an issue, it's going to be hard to figure out what's actually wrong.

It looks just like the other test to me, I actually don't understand why this one doesn't make an issue, and the other one does.

Reply:
Actually I thought about this. 
There are 2 different scenarios when the issues are not created 
1) When the subscriptionGuid id not valid 
2) For the method SynchronizePullRequestAsync. 

For both the scenarios no issue is created a generic method that will not create issue for reasons mentioned above. I did not want to 2 different methods that does the same thing. Felt like it was redundant. Will a comment at the top of the method be good enough? Otherwise I will have to create 2 methods which technically does the same thing.

Reply:
Just a comment would work then.  Or maybe add a parameter of "reason", and all it does it write to a TestOutputHelper.  That way if a task failes, you can read the reason in the test result to see which one went wrong.

Reply:
added comments to explain why certain issue fails.